wasm/libs/default_executor/src/multisig/transaction.rs

```diff
@@ -0,0 +1,260 @@
+};
+
+use super::*;
+
+impl VisitExecute for MultisigPropose {
+    fn visit(&self, executor: &mut Executor) {
+        let host = executor.host();
+        let proposer = executor.context().authority.clone();
+        let target_account = self.account.clone();
+        let multisig_role = multisig_role_for(&target_account);
+        let instructions_hash = HashOf::new(&self.instructions);
+        let is_top_down_proposal = host
+            .query_single(FindAccountMetadata::new(
+                proposer.clone(),
+                SIGNATORIES.parse().unwrap(),
+            ))
+            .map_or(false, |proposer_signatories| {
+                proposer_signatories
+                    .try_into_any::<BTreeMap<AccountId, u8>>()
+                    .dbg_unwrap()
+                    .contains_key(&target_account)
+            });
+
+        if !is_top_down_proposal {
+            let Ok(_role_found) = host
+                .query(FindRolesByAccountId::new(proposer))
+                .filter_with(|role_id| role_id.eq(multisig_role))
+                .execute_single()
+            else {
+                deny!(executor, "not qualified to propose multisig");
+            };
+        }
+
+        let Err(_proposal_not_found) = host.query_single(FindAccountMetadata::new(
+            target_account.clone(),
+            approvals_key(&instructions_hash),
+        )) else {
+            deny!(executor, "multisig proposal duplicates")
+        };
+    }
+
+    fn execute(self, executor: &Executor) -> Result<(), ValidationFail> {
+        let host = executor.host().clone();
+        let proposer = executor.context().authority.clone();
+        let target_account = self.account;
+        let instructions_hash = HashOf::new(&self.instructions);
+        let signatories: BTreeMap<AccountId, u8> = host
+            .query_single(FindAccountMetadata::new(
+                target_account.clone(),
+                SIGNATORIES.parse().unwrap(),
+            ))
+            .dbg_unwrap()
+            .try_into_any()
+            .dbg_unwrap();
+
+        // Recursively deploy multisig authentication down to the personal leaf signatories
+        for signatory in signatories.keys().cloned() {
+            let is_multisig_again = host
+                .query(FindRoleIds)
+                .filter_with(|role_id| role_id.eq(multisig_role_for(&signatory)))
+                .execute_single_opt()
+                .dbg_unwrap()
+                .is_some();
+
+            if is_multisig_again {
+                let propose_to_approve_me = {
+                    let approve_me =
+                        MultisigApprove::new(target_account.clone(), instructions_hash.clone());
+
+                    MultisigPropose::new(signatory, [approve_me.into()].to_vec())
+                };
+                let mut executor = executor.clone();
+                executor.context_mut().authority = target_account.clone();
+                propose_to_approve_me.execute(&executor)?;
```

Comment:
`context_mut` here is only to indicate what authority is executing the instruction:

I've just realized that `host.submit()` goes through no validation, which seems to me to be rather a correct behavior. That's why context mutations whose purpose is to pass permission validation are removed at 5b0dbeb38c1c3ce0a8bd6a517acc142fbe0c4fdd